### PR TITLE
Implement TestContainers Unit Tests with CaptureProxy

### PR DESCRIPTION
**Pull request by AndreKurait**
### Description

Adds CaptureProxy test container tests with a framework to test and simulate a variety of failure modes. In this version, adding the base functionality to spin up docker containers for Kafka as well as a Destination container which are reused between tests, as well as a ToxiProxy container to be used on each test as a proxy between the CaptureProxy and Kafka to enable simulation of failure modes (network disconnect, latency, etc.).

This PR includes a set of tests which simulate Kafka failure modes occurring prior to the start of the CaptureProxy. These tests are currently failing and marked as Disabled due to a bug in the CaptureProxy and will be enabled with the bug fix in [MIGRATIONS-1489](https://opensearch.atlassian.net/browse/MIGRATIONS-1489)

Also, updates test containers version to 1.19.5 to address a mend security issue.

* Category: Enhancement, Maintenance
* Why these changes are required?
     * To enable high velocity testing for Capture Proxy reliability by allowing failure mode simulation environments to be spun up and used in unit tests.
* What is the old behavior before changes and new behavior after changes?
    * No behavior changes

### Issues Resolved
[MIGRATIONS-1503](https://opensearch.atlassian.net/browse/MIGRATIONS-1503)

Is this a backport? If so, please add backport PR # and/or commits #
No

### Testing
Unit Testing

### Check List
- [x ] New functionality includes testing
  - [ x] All tests pass, including unit test, integration test and doctest
- [x ] New functionality has been documented
- [x ] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: TrafficCapture/trafficCaptureProxyServer/src/test/java/org/opensearch/migrations/trafficcapture/proxyserver/ContainerTestBase.java [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+import org.junit.jupiter.api.BeforeEach;
+import org.testcontainers.containers.GenericContainer;
+import org.testcontainers.containers.KafkaContainer;
+import org.testcontainers.containers.ToxiproxyContainer;
+import org.testcontainers.junit.jupiter.Testcontainers;
+import org.testcontainers.lifecycle.Startables;
+import org.testcontainers.utility.DockerImageName;
+
+@Testcontainers(disabledWithoutDocker = true)
+public class ContainerTestBase {
```
Comment (gregschohn) [inline TrafficCapture/trafficCaptureProxyServer/src/test/java/org/opensearch/migrations/trafficcapture/proxyserver/ContainerTestBase.java:16]: Why is this a separate class from the subclass?  The kafka parameters seem weird in a class named "ContainerTestBase"

File: TrafficCapture/trafficCaptureProxyServer/src/test/java/org/opensearch/migrations/trafficcapture/proxyserver/KafkaConfigurationProxyTest.java [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+      assertBasicCalls(captureProxy, 3);
+
+      var averageBaselineDuration = assertBasicCalls(captureProxy, numberOfTests);
+
+      failureMode.apply(kafkaProxy);
+
+      // Calculate average duration of impaired calls
+      var averageImpairedDuration = assertBasicCalls(captureProxy, numberOfTests);
+
+      long acceptableDifference = Duration.ofMillis(25).toMillis();
```
Comment (gregschohn) [inline TrafficCapture/trafficCaptureProxyServer/src/test/java/org/opensearch/migrations/trafficcapture/proxyserver/KafkaConfigurationProxyTest.java:81]: Do you think that this a risk to become a flaky test?  How can we make it relatively stable (> 99-99.9%)?

**Review by gregschohn — APPROVED** [review]

File: TrafficCapture/trafficCaptureProxyServer/src/test/java/org/opensearch/migrations/trafficcapture/proxyserver/KafkaConfigurationProxyTest.java (hunk 2 above)
Comment (AndreKurait) [reply TrafficCapture/trafficCaptureProxyServer/src/test/java/org/opensearch/migrations/trafficcapture/proxyserver/KafkaConfigurationProxyTest.java:81]: The observed latency is 5-10ms on all the runs I've done, that's where 25 was set to have the buffer.

Even when the machine is overloaded by concurrent tests, this is measuring the difference with vs without the proxy so it has some resiliency there. We could move it to a % of the total time if that seems more robust.

File: TrafficCapture/trafficCaptureProxyServer/src/test/java/org/opensearch/migrations/trafficcapture/proxyserver/ContainerTestBase.java (hunk 1 above)
Comment (AndreKurait) [reply TrafficCapture/trafficCaptureProxyServer/src/test/java/org/opensearch/migrations/trafficcapture/proxyserver/ContainerTestBase.java:16]: Good point. I'll split this up and make it more generic

File: TrafficCapture/trafficCaptureProxyServer/src/test/java/org/opensearch/migrations/trafficcapture/proxyserver/KafkaConfigurationProxyTest.java (hunk 2 above)
Comment (AndreKurait) [reply TrafficCapture/trafficCaptureProxyServer/src/test/java/org/opensearch/migrations/trafficcapture/proxyserver/KafkaConfigurationProxyTest.java]: If we see it flaky, we can add a resource lock on it to execute in isolation

**Pull request merged**